clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/id/DcosSpinnakerAppId.groovy

```diff
@@ -0,0 +1,153 @@
+package com.netflix.spinnaker.clouddriver.dcos.deploy.util.id
+
+import static com.google.common.base.Preconditions.checkArgument
+import static com.google.common.base.Strings.nullToEmpty
+
+import com.netflix.frigga.Names
+
+/**
+ * Represents a hierarchical Spinnaker specific application identifier for DCOS.
+ * Structure - /account/region/app-stack-detail-sequence
+ */
+class DcosSpinnakerAppId {
+    public final static def SAFE_REGION_SEPARATOR = "_"
+
+    private final def marathonPath
+
+    public DcosSpinnakerAppId(final String marathonAppId) {
```

Comment:
I don't understand the need for static constructors just to describe what they were coming from (considering the number of parameters gives that away). I can keep that nomenclature for the existing static constructors (which return optionals instead of the actual thing, as a safer method of creating an AppId) but I don't understand the need to not allow public constructors.